src/kernel/include/thd.h

```diff
@@ -192,9 +193,32 @@ thd_rcvcap_evt_dequeue(struct thread *head) { return list_dequeue(&head->event_h
 static inline int
 thd_track_exec(struct thread *t) { return !list_empty(&t->event_list); }
 
+static void
+thd_rcvcap_all_pending_set(struct thread *t, int val)
+{ t->rcvcap.is_all_pending = val; }
+
+static int
+thd_rcvcap_all_pending_get(struct thread *t)
+{ return t->rcvcap.is_all_pending; }
+
+static int
+thd_rcvcap_all_pending(struct thread *t)
+{
+	int pending = t->rcvcap.pending;
+
+	/* receive all pending */
+	t->rcvcap.pending = 0;
+	thd_rcvcap_all_pending_set(t, 0);
+
+	return ((pending<<1) | (list_first(&t->event_head) != NULL));
```

Comment:
I think that what's making it slightly confusing is the `!=` and conditional.

Better might be `return ((pending << 1) | list_empty(&t->event_head));`  Should we add that function?

Notice the extra spaces around the binary operator change here as well (`pending<<1` vs. `pending << 1`).